nav2_util/src/robot_utils.cpp

```diff
@@ -28,39 +28,47 @@ bool getCurrentPose(
   tf2_ros::Buffer & tf_buffer, const std::string global_frame,
   const std::string robot_frame, const double transform_timeout)
 {
-  static rclcpp::Logger logger = rclcpp::get_logger("getCurrentPose");
-  geometry_msgs::msg::PoseStamped robot_pose;
-
   tf2::toMsg(tf2::Transform::getIdentity(), global_pose.pose);
-  tf2::toMsg(tf2::Transform::getIdentity(), robot_pose.pose);
-  robot_pose.header.frame_id = robot_frame;
-  robot_pose.header.stamp = rclcpp::Time();
+  global_pose.header.frame_id = robot_frame;
+  global_pose.header.stamp = rclcpp::Time();
+
+  return transformPoseInTargetFrame(
+    global_pose, global_pose, tf_buffer, global_frame, transform_timeout);
+}
+
+bool transformPoseInTargetFrame(
```

Comment:
With the new API for `transformPoseToGlobalFrame`, wouldn't you only need the original function that was here?

Reply:
First, yes, tested and working on a real robot.

Second, I am not sure I am following. I see many different ways of doing this, but I don't see how to do it without keeping both `nav2_util::transformPoseInTargetFrame` and `nav2_util::getCurrentPose`.